### DISCUSSION TITLE
GCC 7.1 Compilation warning when compiling science and arm apps

### DISCUSSION THREAD

**Discussion by jessieramseyer** [uwrobotics/uwrt_mars_rover#158] [Q&A]
![image](https://user-images.githubusercontent.com/54551825/137194839-26737572-2424-4e3d-90d9-7f4f2078bd53.png)                     

I am making my way through the firmware training document. When I tried compiling the science and arm apps I receive these warnings.

**Comment by cindyli-13** [comment]
Did the apps still compile? This could just be a warning that doesn't affect compilation

**Reply by jessieramseyer** [reply]
![image](https://user-images.githubusercontent.com/54551825/137226807-b2c8201e-45e9-4915-b8be-dc0dc3f7b53d.png)
I rebooted my WSL terminal and the warnings have disappeared...? This doesn't match the expected output from the readme. Do you know if this is okay too?

**Reply by cindyli-13** [reply]
Could be because the app was already built? If you find under the folder `cmake_build/SCIENCE_REV2_2021/develop/GCC_ARM/rover-apps/science_2021` a `.bin` and `.elf` and `.elf.map` and `.hex` file then that means you've already built the app

**Reply by wmmc88** [reply]
To echo what @cindyli-13 says, the `ninja: no work to do` indicates that the target has already been built.


Theres a `clean` target in the makefile that u can use to delete the built artifacts (ie. `make clean`)

**Comment by wmmc88** [comment]
> ![image](https://user-images.githubusercontent.com/54551825/137194839-26737572-2424-4e3d-90d9-7f4f2078bd53.png)                     
> 
> I am making my way through the firmware training document. When I tried compiling the science and arm apps I receive these warnings. 
> 

Just fyi for github etiquette, always post terminal output using markdown code blocks instead of images so that its searchable.
